### ISSUE TITLE
[Bug]: pinecone-client version too old, didn't match with langchain-pinecone

### ISSUE THREAD

**Issue by upchunk** [run-llama/llama_index#15120]
### Bug Description

Installing llama-index-vector-store-pinecone along with langchain-pinecone caused the langchain-pinecone and pinecone-client version to be downgraded 
![image](https://github.com/user-attachments/assets/f5899a0c-559b-4194-9512-53a05ec2b512)

### Version

0.10.59

### Steps to Reproduce

1. install langchain, langchain-community, langchain-pinecone, pinecone-client
2. install llama-index, llama-index-vector-store-pinecone

### Relevant Logs/Tracbacks

```shell
Using cached langchain_pinecone-0.1.2-py3-none-any.whl (8.5 kB)
Using cached pinecone_client-3.2.2-py3-none-any.whl (215 kB)
Installing collected packages: pinecone-client, langchain-pinecone
  Attempting uninstall: pinecone-client
    Found existing installation: pinecone-client 5.0.1
    Uninstalling pinecone-client-5.0.1:
      Successfully uninstalled pinecone-client-5.0.1
  Attempting uninstall: langchain-pinecone
    Found existing installation: langchain-pinecone 0.1.3
    Uninstalling langchain-pinecone-0.1.3:
      Successfully uninstalled langchain-pinecone-0.1.3
Successfully installed langchain-pinecone-0.1.2 pinecone-client-3.2.2
```

**Comment by upchunk** [comment]
I need the pinecone-client>4.0.0

**Comment by upchunk** [comment]
I Fix it by my self #15121
